hantec_api_ecommerce/controllers/main.py

```diff
@@ -0,0 +1,817 @@
+            "contact_id": new_contact.id,
+        }
+
+    @route(
+        '/update_contact/<model("res.partner"):partner>',
+        methods=["POST"],
+        type="json",
+        auth="user",
+    )
+    def update_contact(self, partner=False):
+        """Updates a contact with new values.
+
+        This function updates the details of a specified contact using the values
+        provided in the JSON request body.
+
+        JSON request body:
+            - partner_id (int): The ID of the contact to be updated.
+            - update_vals (dict): A dictionary containing the values to be updated.
+
+        JSON response:
+            - message (str): A message indicating that the contact has been successfully updated.
+
+        Returns:
+            dict: A dictionary with a success message.
+
+        """
+        update_vals = request.jsonrequest.get("update_vals")  # Values to update
+        partner.write(update_vals)  # Update the contact with the new values
+        logger.debug("Contact updated with ID %s", partner.id)
+
+        return {"message": f"Contact with ID: {partner.id} successfully updated."}
+
+    @route("/address_invoice", methods=["POST"], type="json", auth="user")
+    def create_address_invoice(self):
+        """Creates or updates the billing address for a given partner.
+
+        This function creates or updates the billing address of a specified partner
+        using the details provided in the JSON request body.
+
+        JSON request body:
+            - partner_id (int): The ID of the partner.
+            - invoice_data (dict): A dictionary containing the billing address details.
+
+        JSON response:
+            - message (str): A message indicating that the billing address has been successfully created or updated.
+            - invoice_id (int): The ID of the created or updated billing address.
+
+        Returns:
+            dict: A dictionary with a success message and the billing address ID.
+
+        """
+        env = request.env
+        data = request.jsonrequest
+        partner_id = data.get("partner_id")
+        invoice_data = data.get("invoice_data")
+
+        # Search if a billing address already exists
+        invoice_address = env["res.partner"].search(
+            [("parent_id", "=", partner_id), ("type", "=", "invoice")], limit=1
+        )
+
+        if invoice_address:
+            # Update existing billing address
+            invoice_address.write(invoice_data)
+        else:
+            # Add a new billing address
+            invoice_data.update({"type": "invoice", "parent_id": partner_id})
+            invoice_address = env["res.partner"].create(invoice_data)
+
+        return {
+            "message": "Billing address successfully updated/created.",
+            "invoice_id": invoice_address.id,
+        }
+
+    @route("/delivery_address", methods=["POST"], type="json", auth="user")
+    def delivery_address(self):
```

Comment:
suggestion:

you could merge this endpoint at the one above at create_address_invoice

it's the same code, just here are handaling the only_create boolean

my recomendation is to merge them in a sigle method